pkg/yurthub/cachemanager/cache_manager.go

```diff
@@ -146,21 +146,27 @@ func (cm *cacheManager) queryListObject(req *http.Request) (runtime.Object, erro
 		return nil, err
 	}
 
-	var gvk schema.GroupVersionKind
 	var kind string
 	objs, err := cm.storage.List(key)
-	if err != nil {
-		return nil, err
-	} else if len(objs) == 0 {
-		gvk, err = serializer.UnsafeDefaultRESTMapper.KindFor(schema.GroupVersionResource{
+	// storage.ErrStorageNotFound indicates that there is no corresponding resource locally,
+	// and will determine whether to return an empty list structure according to whether GVK is registered or not
+	if err == storage.ErrStorageNotFound {
+		gvr := schema.GroupVersionResource{
 			Group:    info.APIGroup,
 			Version:  info.APIVersion,
 			Resource: info.Resource,
-		})
-		if err != nil {
-			return nil, err
+		}
+		gvk, e := serializer.YurtHubRESTMapperManager.UnsafeDefaultRESTMapper.KindFor(gvr)
```

Comment:
It's not good to expose `RESTMaaper` info out of serializer, how about add a func `KindFor()` for serializer?

Reply:
I combined RESTMapperManager with CacheManager, and the interface is exposed by CacheManager